grc: Fix YAML hier blocks (next)

haakov opened the pull request:
Switched to YAML for hierarchical blocks

haakov commented on grc/core/generator/top_block.py:
```diff
@@ -234,15 +234,20 @@ def make_port_sig(port):
             connections.remove(connection)
 
+        for connection in virtual_sink_connections:
+            connections.remove(connection)
```
This prevents lines like `self.connect((self.analog_sig_source_x_0, 0), (self.virtual_sink_0, 0))` from appearing in the Python files. Attempting to use connect() with a virtual source/sink creates an `AttributeError: 'top_block_sptr' object has no attribute 'virtual_sink_0'`.

marcusmueller commented:
so, following the discussion on #1886, we should probably be merging this on next, right? Or did we already?

haakov commented:
This hasn’t been merged on next yet. I actually don’t think this PR has been touched since I created it in November, so because of that I’d like a couple of days to make sure it works if that’s okay. I’ll notify you when it’s been tested!

haakov commented:
Okay so the first commit looks fine, but the one regarding virtual connections does not work with `next`. I'll have a look at it.

haakov commented:
@marcusmueller It looks like the virtual connections are going to take some time to figure out, so I removed the commit that didn't work and squashed the others. This way you can merge this PR whenever you'd like and don't have to wait for me to figure out how to fix the virtual connections :)

marcusmueller commented:
Thanks for the incredible patience! And, even more so, the incredible work :)
